### PR TITLE
tail: don't optimize beyond fetch_size

### PR DESCRIPTION
**Pull request by davidvgalbraith**
If the tail limit exceeds the fetch size, then we don't get points
in the right order. Rather than add a bunch of complexity to keep
optimizing this edge case, let's just not optimize it.

@VladVega

### CODE REVIEW

File: test/elastic-adapter-limits.js [hunk 1]
```diff
@@ -62,13 +62,13 @@ describe('elastic source limits', function() {
...
 
-            it('enforces tail across multiple fetches', function() {
-                var extra = '-fetch_size 2 | tail 4';
+            it('doesn\'t optimize tail in excess of fetch size', function() {
+                var extra = '-fetch_size 2 | tail 8';
                 return test_utils.read({id: type}, extra)
                 .then(function(result) {
-                    var expected = _.last(points, 4);
+                    var expected = _.last(points, 8);
                     test_utils.check_result_vs_expected_sorting_by(result.sinks.table, expected, 'bytes');
```
Comment (VladVega) [inline test/elastic-adapter-limits.js]: I still don't really get why we are re-sorting the data here.

Comment (davidvgalbraith) [inline test/elastic-adapter-limits.js]: The last 4 points are simultaneous, so they come back from Juttle in a different order from our data file.

File: lib/optimize.js [hunk 2]
```diff
@@ -76,6 +76,12 @@ var optimizer = {
             limit = Math.min(limit, optimization_info.limit);
         }
 
+        var read_fetch_size = graph.node_get_option(read, 'fetch_size');
```
Comment (VladVega) [inline lib/optimize.js]: what if `fetch_size` is not indicated in the juttle? Consider using the default value?

File: test/elastic-adapter-limits.js (hunk 1 above)
Comment (VladVega) [inline test/elastic-adapter-limits.js]: I see. So then I noticed that all your tail unit tests execute `check_result_vs_expected_sorting_by` so they never verify correct order. Which means you could probably not perform the reverse and all the tail tests would pass. Is that correct?

File: lib/optimize.js (hunk 2 above)
Comment (davidvgalbraith) [inline lib/optimize.js]: Great idea! Updated.

File: test/elastic-adapter-limits.js (hunk 1 above)
Comment (davidvgalbraith) [inline test/elastic-adapter-limits.js]: They do! Check `check_result_vs_expected_sorting_by`.

Comment (VladVega) [inline test/elastic-adapter-limits.js]: But not sorted the same as unoptimized.
```expect_sorted(received);
expect(_.sortBy(received, field)).deep.equal(_.sortBy(expected, field));```

Comment (VladVega) [inline test/elastic-adapter-limits.js]: I guess maybe that's ok... I guess if you think that's ok, then +1 on PR.

**Comment by VladVega** [comment]
Just in case you didn't see my threaded +1 above, then +1.
